manifests/cli/config.pp

```diff
@@ -24,8 +24,10 @@
   validate_string($ssh_private_key_content)
 
   # required by PuppetX::Jenkins::Provider::Clihelper base
-  package { 'retries':
-    provider => 'gem',
+  if ! defined(Package['retries']) {
```

Comment:
@boyand There has been some discussion of puppet "slotting" resources by provider specifically because of the 3 different providers for needed for "unvendored", puppetserver, & puppet-agent.  I'm not sure what the correct thing to do here is -- perhaps remove the resource completely and make it the end-user's responsibility?